src/lib/utils.sh

```diff
@@ -44,27 +55,30 @@ function parallel-scp() {
   host="$2"
   path="$3"
 
-  if [[ ! -f "${f}.sig" ]]; then
+  pushd "$(dirname "$f")"
```

Comment:
Now that we have a pushd here, it's better that in deploy function, this one is called using a subshell, this way if it fails before popd, the current PWD is discarded...